ckan/views/dataset.py

```diff
@@ -754,6 +754,12 @@ def get(
         pkg = context.get(u"package")
         resources_json = h.json.dumps(data.get(u'resources', []))
 
+        # if dataset was created via API, license_id is None
+        # set it to nonspecified to avoid accidental update
```

Comment:
Does this really prevent accidental updates? What about cases, when custom licenses are used instead of built-in ones and there is no 'notspecified' option?

Reply:
Yes, I understand now. So what is the best approach to fix it? Add additional field `notspecified`  if user list of licenses was provided? Fix in `select2.js` to prevent accidental choose? Make license mandatory in package_create API method? Or ..?

Reply:
As i said a month ago:
> We've discussed this problem on tech meeting - a much better solution would be to remove/map explicit `not specified` to an empty value and display empty license as Not Specified.